pkg/gather/clusterconfig/pod_network_connectivity_checks.go

```diff
@@ -0,0 +1,90 @@
+package clusterconfig
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	controlplanev1 "github.com/openshift/api/operatorcontrolplane/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherPNCC collects a summary of failed PodNetworkConnectivityChecks.
+// The checks are requested via a dynamic client and
+// then unmarshaled into the appropriate structure.
+//
+// Resource API: podnetworkconnectivitychecks.controlplane.operator.openshift.io/v1alpha1
+// Docs for relevant types: https://pkg.go.dev/github.com/openshift/api/operatorcontrolplane/v1alpha1
+//
+// * Location in archive: config/podnetworkconnectivitychecks.json
+// * Id in config: pod_network_connectivity_checks
+// * Since versions:
+//   * 4.8+
+func GatherPNCC(g *Gatherer, c chan<- gatherResult) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+
+	records, errors := gatherPNCC(g.ctx, gatherDynamicClient, gatherKubeClient.CoreV1())
+	c <- gatherResult{records: records, errors: errors}
+}
+
+func getUnsuccessfulChecks(entries []controlplanev1.LogEntry) []controlplanev1.LogEntry {
+	unsuccesseful := []controlplanev1.LogEntry{}
+	for _, entry := range entries {
+		if !entry.Success {
+			unsuccesseful = append(unsuccesseful, entry)
+		}
+	}
+	return unsuccesseful
+}
+
+func gatherPNCC(ctx context.Context, dynamicClient dynamic.Interface, coreClient corev1client.CoreV1Interface) ([]record.Record, []error) {
```

Comment:
Linter told me that `coreClient` was not used :)

Reply:
VSC betrayed me and didn't notice that. I'm surprised it even compiled with an unreferenced variable.